plugins/cache-ehcache/src/test/java/org/opensearch/cache/store/disk/EhcacheThreadLeakFilter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cache.store.disk;
+
+import com.carrotsearch.randomizedtesting.ThreadFilter;
+
+/**
+ * In Ehcache(as of 3.8.0), while calling remove/invalidate() on entries causes to start a daemon thread in the
+ * background to clean up the stale offheap memory associated with the disk cache. And this thread is not closed even
+ * after we try to close the cache or cache manager. Considering that it requires a node restart to switch between
+ * different cache plugins, this shouldn't be a problem for now. Will be following up with ehcache on this.
+ */
+public class EhcacheThreadLeakFilter implements ThreadFilter {
+
+    private static final String OFFENDING_THREAD_NAME = "MappedByteBufferSource";
+
+    @Override
+    public boolean reject(Thread t) {
+        return t.getName().startsWith(OFFENDING_THREAD_NAME);
```

Comment:
@sgup432 where this thread is coming from? how could we shutdown EhCache in a way that these threads are terminated?

Reply:
Thanks @sgup432 , I suspect this is https://github.com/ehcache/ehcache3/issues/3204 ? Could you please add a reference to `EhcacheThreadLeakFilter` why we are doing this? Thank you.

Reply:
Yeah sure. That is the issue within ehcache3 which I had opened after talking with them offline via google group :D .
I will attach this link.

Reply:
Thank you @sgup432 !